fs/crypto/bio.c

```diff
@@ -131,7 +131,7 @@ int fscrypt_zeroout_range(const struct inode *inode, pgoff_t lblk,
 		bio->bi_bdev = inode->i_sb->s_bdev;
 		bio->bi_iter.bi_sector =
 			pblk << (inode->i_sb->s_blocksize_bits - 9);
-		bio_set_op_attrs(bio, REQ_OP_WRITE, REQ_NOENCRYPT);
```

Comment:
Your commit message doesn't do a great job of explaining why this change needs to be undone. There are a few instances in the kernel where a flag other than 0 is passed with `REQ_OP_WRITE`:

```bash
$ rg -F "bio_set_op_attrs(bio, REQ_OP_WRITE" | rg -v 0
block/blk-flush.c:      bio_set_op_attrs(bio, REQ_OP_WRITE, WRITE_FLUSH);
mm/page_io.c:           bio_set_op_attrs(bio, REQ_OP_WRITE, REQ_SYNC);
fs/btrfs/disk-io.c:     bio_set_op_attrs(bio, REQ_OP_WRITE, WRITE_FLUSH);
fs/btrfs/extent_io.c:   bio_set_op_attrs(bio, REQ_OP_WRITE, WRITE_SYNC);
fs/btrfs/scrub.c:       bio_set_op_attrs(bio, REQ_OP_WRITE, WRITE_SYNC);
fs/crypto/bio.c:                bio_set_op_attrs(bio, REQ_OP_WRITE, REQ_NOENCRYPT);
drivers/md/md.c:        bio_set_op_attrs(bio, REQ_OP_WRITE, WRITE_FLUSH_FUA);
fs/jfs/jfs_logmgr.c:    bio_set_op_attrs(bio, REQ_OP_WRITE, WRITE_SYNC);
drivers/target/target_core_iblock.c:    bio_set_op_attrs(bio, REQ_OP_WRITE, WRITE_FLUSH);
fs/sdfat/mpage.c:       bio_set_op_attrs(bio, REQ_OP_WRITE, flags);
drivers/md/bcache/journal.c:            bio_set_op_attrs(bio, REQ_OP_WRITE,
drivers/md/bcache/super.c:      bio_set_op_attrs(bio, REQ_OP_WRITE, REQ_SYNC|REQ_META);
drivers/nvme/target/io-cmd.c:   bio_set_op_attrs(bio, REQ_OP_WRITE, WRITE_FLUSH);
```

Furthermore, the ext4 maintainer +1'd this change on Gerrit, I would think it was correct: https://android-review.googlesource.com/c/kernel/common/+/593373

Reply:
Ok